src/main/java/org/ngafid/events/proximity/FlightTimeLocation.java

```diff
@@ -63,22 +64,20 @@ public FlightTimeLocation(Connection connection, Flight flight) throws SQLExcept
         Pair<Double, Double> minMaxRPM2 = DoubleTimeSeries.getMinMax(connection, flightId, "E2 RPM");
 
 
-        //Parquet files do not have rpm data, skip this check.
-        if (!flight.getFilename().endsWith(".parquet")) {
-            if ((minMaxRPM1 == null && minMaxRPM2 == null)
-                    || (minMaxRPM2 == null && minMaxRPM1.second() < 800)
-                    || (minMaxRPM1 == null && minMaxRPM2.second() < 800)
-                    || ((minMaxRPM1 != null && minMaxRPM2 != null && minMaxRPM1.second() < 800
-                    && minMaxRPM2.second() < 800))) {
-                valid = false;
-                return;
-            }
+        // Invalidate the flight if RPM is present and too low
+        if ((minMaxRPM1 != null && minMaxRPM1.second() < 800) &&
+                (minMaxRPM2 != null && minMaxRPM2.second() < 800)) {
```

Comment:
We only need both RPM to be < 800 if there are two RPM values. You could replace the second part of the && to be `minMaxRPM == null || minMaxRPM2.second() < 800`, since this will only require the second RPM to be < 800 if it exists.